Benqi-Agents/Qi-Transfers/src/data.fetcher.ts

```diff
@@ -1,20 +1,28 @@
 import { Contract, providers, BigNumber } from "ethers";
 import { benqiInterface } from "./utils";
+import LRU from "lru-cache";
 
 export default class DataFetcher {
   readonly qiTokenAddress: string;
   readonly provider: providers.Provider;
   private qiTokenContract: Contract;
+  private cache: LRU<string, BigNumber>;
 
   constructor(qiTokenAddress: string, provider: providers.Provider) {
     this.qiTokenAddress = qiTokenAddress;
     this.provider = provider;
     this.qiTokenContract = new Contract(qiTokenAddress, benqiInterface, provider);
+    this.cache = new LRU<string, BigNumber>({ max: 10000 });
   }
 
   // return the number of tokens held by the account
   public async getBalance(account: string, blockNumber: number): Promise<BigNumber> {
+    const key: string = `${account + blockNumber}`;

```

Comment:
```suggestion
    const key: string = `${account}-${blockNumber}`;
```
`string + number` doesn't look like a good practice for me. Please consider to apply this suggestion.